src/main/java/com/educare/unitylend/service/impl/BorrowRequestServiceImpl.java

```diff
@@ -0,0 +1,59 @@
+package com.educare.unitylend.service.impl;
+
+import com.educare.unitylend.Exception.ServiceException;
+import com.educare.unitylend.dao.BorrowRequestRepository;
+import com.educare.unitylend.dao.UserRepository;
+import com.educare.unitylend.model.BorrowRequest;
+import com.educare.unitylend.model.User;
+import com.educare.unitylend.service.BorrowRequestService;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.List;
+
+@Slf4j
+@AllArgsConstructor
+@Service
+public class BorrowRequestServiceImpl implements BorrowRequestService {
+    private UserRepository userRepository;
+    private BorrowRequestRepository borrowRequestRepository;
+    public List<String> getRequestedCommunitiesByUserId(String userId) throws ServiceException{
+        List<String> communities=borrowRequestRepository.getBorrowRequestByUserId(userId);
+        System.out.println(communities);
+        return communities;
+    }
+
+    public void createBorrowRequest(BorrowRequest borrowRequest) {
+        borrowRequestRepository.createBorrowRequest(borrowRequest);
+    }
+    public boolean raiseBorrowRequestByUserId(String userId, BorrowRequest borrowRequest) throws ServiceException {
+        try {
+            // Assuming returnPeriod is in days
+            BigDecimal monthlyInterestRate = new BigDecimal("50.0").divide(new BigDecimal(12 * 100), 10, RoundingMode.HALF_UP);
```

Comment:
1. Move all the constants used to the start of the class, i.e ` public static final Integer MONTHS_IN_YEAR = 12.`
2. Add interest_rate in the BorrowRequest Object
3. Add loan tenure in BorrowRequest object. 
4. The validation method here should just calculate the monthly EMI and then validate based on the user income if the user will be able to pay the amount or not, By checking if the user income is double of the EMI or not.

Reply:
And for calculating the EMI you should have different function calculateEMI(PrincipalAmount, loanPaymentTenure, InterestRate) and this function should utilize that.